src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -389,24 +399,45 @@ public static StarTable getStarTable(String fits_filename, int table_idx) {
 
         FitsFactory.useThreadLocalSettings(true);
         FitsFactory.setLongStringsEnabled(false);
+        Fits fits = null;
 
         try {
             StarTableFactory stFactory = new StarTableFactory();
+            StarTable tbl = null;
 
             int tIdx = table_idx < 1 ? 1 : table_idx;
 
-            StarTable tbl = stFactory.makeStarTable(fits_filename + "#" + tIdx, "fits");
-
-            if (tbl != null) {
-                DescribedValue dValue = tbl.getParameterByName("ZIMAGE");   // check if 'BINTABLE' is actually compressed image HDU
+            fits = new Fits(fits_filename);
+            BasicHDU[] parts = fits.read();
+            if (tIdx>=parts.length) {
+                logger.error("unable to get table from fits file " + fits_filename +
+                        "from" + table_idx + ", reason: HDU does not exist, HDU count: "+ parts.length);
+                return null;
+            }
+            BasicHDU hdu= parts[tIdx];
+            if (hdu instanceof TableHDU) {
+                tbl= new FitsStarTable((TableHDU)parts[tIdx]);
+            }
+            else {
+                logger.error("unable to get table from fits file " + fits_filename +
+                        "from" + table_idx + ", reason: HDU is not a table hdu");
+                return null;
+            }
+            DescribedValue dValue = tbl.getParameterByName("ZIMAGE");   // check if 'BINTABLE' is actually compressed image HDU
 
-                if (dValue == null || dValue.getValue() != Boolean.TRUE) {
-                    retTbl = tbl;
-                }
+            if (dValue == null || dValue.getValue() != Boolean.TRUE) {
+                retTbl = tbl;
             }
         } catch (Exception e) {
             logger.error("unable to get table from fits file " + fits_filename + "from" + table_idx + ", reason: " + e.getMessage());
         }
+        finally {
+            try {
+                if (fits!=null && fits.getStream()!=null) fits.getStream().close();
+            } catch (IOException e) {
+                // do nothing
+            }
+        }
 
         FitsFactory.useThreadLocalSettings(false);
```

Comment:
looking over the code this morning. I think i will move 
```java
 FitsFactory.useThreadLocalSettings(false);
```
to the finally block